### ISSUE TITLE
Running the action twice fails

### ISSUE THREAD

**Issue by isc30** [JamesIves/github-pages-deploy-action#284]
I'm trying to use the action to perform 2 actions:
1. Publish a folder to gh-pages branch in my repo
2. Publish another folder (wiki) to MyRepo.wiki (another repo)

When I perform any of them separately, everything works fine but when I try to execute both as part of the same Action, it crashes with the following message:

```
Checking configuration and starting deployment… 🚦
Deploying using Access Token… 🔑
Configuring git…
##[error]There was an error initializing the repository: The process '/usr/bin/git' failed with exit code 128 ❌
Deployment Failed ❌
```

You can see the failing action here: https://github.com/isc30/blazor-lazy-loading/runs/658932569

And the workflow file here: https://github.com/isc30/blazor-lazy-loading/actions/runs/99942724/workflow#L65

**Comment by JamesIves** [comment]
Could you follow the steps provided [here](https://github.com/JamesIves/github-pages-deploy-action#debugging-) and run the workflow again? (This reminds me that I should add this to the issue template)

**Comment by isc30** [comment]
I cannot do that atm but it can be reproduced by using the action twice in the same file one after another

**Comment by JamesIves** [comment]
Thanks for the repro information. I will investigate this later this week.

**Comment by plimerence** [comment]
same issue

**Comment by JamesIves** [comment]
Sorry this took a little while for me to investigate - @isc30 can you verify that this repository path is correct? `https://github.com/isc30/blazor-lazy-loading.wiki.git`? I'm guessing this repository is private but the `.git` at the end looks strange. I'm also seeing this issue when I cross deploy right after one run to another and it's to do with the temporary subtree the action creates, but it's failing much later in the deployment which makes me wonder if your token is not scoped enough to see the repository you're attempting to deploy to.

`fatal: 'gh-action-temp-deployment-folder' is a missing but already registered worktree;`

I'm going to push up a fix that randomizes this name to prevent this from happening in the future.

**Comment by JamesIves** [comment]
This has been resolved and released as part of patch `3.5.3`, you can find the changelog here: https://github.com/JamesIves/github-pages-deploy-action/releases/tag/3.5.3
